vra/resource_block_device.go

```diff
@@ -249,7 +252,125 @@ func resourceBlockDeviceRead(d *schema.ResourceData, m interface{}) error {
 
 func resourceBlockDeviceUpdate(d *schema.ResourceData, m interface{}) error {
 
-	return fmt.Errorf("Updating a block device resource is not allowed")
+	log.Printf("Starting to update the vra_block_device resource with name %s", d.Get("name"))
+	apiClient := m.(*Client).apiClient
+
+	id := d.Id()
+	if d.HasChange("capacity_in_gb") {
+		err := resizeDisk(d, apiClient, id)
+		if err != nil {
+			return err
+		}
+	}
+
+	log.Printf("Finished updating vra_block_device resource with name %s", d.Get("name"))
+	return nil
+}
+
+func resizeDisk(d *schema.ResourceData, apiClient *client.MulticloudIaaS, id string) error {
+
+	log.Printf("Starting resize of vra_block_device resource with name %s", d.Get("name"))
+	// get the block device by id
```

Comment:
There are some comments in this file that start with a lower case letter. Be good to standardize on capitalizing the start of a comment. Also, there is a "NOTE" that needs an extra space to separate from the comment start.

Reply:
done!